src/api/mod.rs

```diff
@@ -48,17 +53,23 @@ impl Client {
             .post(self.url("/check"))
             .query(request)
             .send()
-            .await?
+            .await
+            .map_err(Error::RequestEncode)?
             .json::<check::Response>()
+            .await
+            .map_err(Error::ResponseDecode)
     }
 
-    /// Send a words request to the server and await for the response.
-    pub async fn languages(&self, request: &languages::Request) -> Result<languages::Response> {
+    /// Send a request for the list of supported languages to the server and
+    /// await for the response.
+    pub async fn languages(&self) -> Result<languages::Response> {
         self.client
             .get(self.url("/languages"))
-            .query(request)
             .send()
-            .await?
-            .json::<check::Response>()
+            .await
+            .map_err(Error::RequestEncode)?
+            .json::<languages::Response>()
+            .await
+            .map_err(Error::ResponseDecode)
```

Comment:
Actually, the removal of the inner `?` shadowed the possibility to return early an error due to a bad reqwest. I think you should to rollback to the original code for this, see my commit suggestions.

Reply:
I still have a `?` for an early return, just after the error is mapped to this crate's error type. I think your suggested way would just convert it into a standard `Reqwest` error instead of that `RequestEncode`, is that what it's supposed to be?

```
    /// Any other error from requests (see [`reqwest::Error`]).
    #[error(transparent)]
    Reqwest(#[from] reqwest::Error),
```

Reply:
Hum, I am not sure to understand. Originally, there was `.send().await?`, which would return early if some `Reqwest::Error` was returned by `.send()`. Because I derive `thiserror::Error` on my error type, any `Reqwest::Error` will be mapped into `Error::Reqwest` (because `?` calls `err.into()`).

Now that you changed to `send().map_err(Error::RequestEncode)`, any `Reqwest::Error` will be transformed into `Error::RequestEncode`, which doesn't make sense because those errors do not mean the same thing.

Reply:
Sure but it contains the `reqwest` error, and is returned early too, but if you're saying it's the incorrect type then that's different yeah. I was just copying the error type from `server.rs`:

https://github.com/Rolv-Apneseth/languagetool-rust/blob/917e6c8473ba81117ad99b887436f99586c60b69/src/api/server.rs#L369C1-L403C6

Reply:
I see, and this would make sense if there was a possibility that the request fails to be built. But I cannot really happen since we know the content of the request. This is not the case for `check`, where a malformed string couldn't be transformed into a proper HTML request, I don't know.

Probably that all request errors should be merged into one, because there are some cases where we return `EncodeError` and that it is misleading about what really happened. But this is probably the work of another PR :-)

Reply:
Makes sense, yeah I think all the `reqwest` errors can be covered by `error::Reqwest` like you suggest. Want me to do anything for now?

Reply:
I'll just accept those changes?

Reply:
You can accept, I'll run checks to make sure it didn't break anything (I did check my code haha), and we can remove `Error::RequestEncode` and `Error::RespondeDecode` in another PR :-)